packages/iota/src/binary/payloads/milestonePayload.ts

```diff
@@ -141,15 +140,16 @@ export function serializeMilestonePayload(writeStream: WriteStream, object: IMil
     writeStream.writeUInt32("payloadMilestone.nextPoWScore", object.nextPoWScore);
     writeStream.writeUInt32("payloadMilestone.nextPoWScoreMilestoneIndex", object.nextPoWScoreMilestoneIndex);
 
-    writeStream.writeUInt8("payloadMilestone.publicKeysCount", object.publicKeys.length);
-    for (let i = 0; i < object.publicKeys.length; i++) {
-        writeStream.writeFixedHex("payloadMilestone.publicKey", Ed25519.PUBLIC_KEY_SIZE, object.publicKeys[i]);
+    const metadata = HexHelper.stripPrefix(object.metadata);
+    writeStream.writeUInt16("payloadMilestone.metadataLength", metadata.length / 2);
```

Comment:
Does the `metadataLength` field need to be written even if `metadata.lenght > 0 == false`, cause like this we always write it. If not, this line could be inside condition on line below.